anesthetic/read/chain.py

```diff
@@ -0,0 +1,29 @@
+"""Read MCMCSamples or NestedSamples from any chains."""
+from anesthetic.read.polychord import read_polychord
+from anesthetic.read.getdist import read_getdist
+from anesthetic.read.cobaya import read_cobaya
+from anesthetic.read.multinest import read_multinest
+
+
+def read_chains(root, *args, **kwargs):
+    """Auto-detect chain type and read from file.
+
+    Parameters
+    ----------
+    root: str
+        root name
+
+    Returns
+    -------
+    NestedSamples or MCMCSamples depending on auto-detection
+
+    """
+    errors = []
+    for read in [read_polychord, read_multinest, read_cobaya, read_getdist]:
```

Comment:
Note the importance of the order here: `read_cobaya` has to come before `read_getdist`.

Reply:
This is subtle. Well spotted.